internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go

```diff
@@ -141,28 +142,45 @@ func (h *PolicyChangeHandler) handleFleetServerHosts(ctx context.Context, c *con
 	// only set protocol/hosts as that is all Fleet currently sends
 	prevProtocol := h.config.Fleet.Client.Protocol
 	prevPath := h.config.Fleet.Client.Path
+	prevHost := h.config.Fleet.Client.Host
 	prevHosts := h.config.Fleet.Client.Hosts
 	prevProxy := h.config.Fleet.Client.Transport.Proxy
 	h.config.Fleet.Client.Protocol = cfg.Fleet.Client.Protocol
 	h.config.Fleet.Client.Path = cfg.Fleet.Client.Path
+	h.config.Fleet.Client.Host = cfg.Fleet.Client.Host
 	h.config.Fleet.Client.Hosts = cfg.Fleet.Client.Hosts
-	h.config.Fleet.Client.Transport.Proxy = cfg.Fleet.Client.Transport.Proxy
+
+	// Empty proxies from fleet are ignored. That way a proxy set by --proxy-url
+	// it won't be overridden by an absent or empty proxy from fleet-server.
+	// However, if there is a proxy sent by fleet-server, it'll take precedence.
+	// Therefore, it's not possible to remove a proxy once it's set.
+	if cfg.Fleet.Client.Transport.Proxy.URL == nil ||
+		cfg.Fleet.Client.Transport.Proxy.URL.String() == "" {
```

Comment:
consider trimming a space to rule out whitespace strings

Reply:
well, empty space gets URL encoded, so trimming won't work. Also the URL (`httpcommon.ProxyURI` type) is created by `url.Parse` when `httpcommon.NewProxyURIFromString` is used. The best place to deal with blank strings would be on `elastic-agent-libs` directly.

Anyway the proxy as well as the fleet-server URLs are tested, so unless a blank string is indeed reachable or works as proxy it won't be applied. Thus I'd say it's safe to let it as it is.